sysid-application/src/test/native/cpp/generation/HardwareTypeTest.cpp

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+#include <functional>
+#include <stdexcept>
+#include <string_view>
+
+#include <wpi/StringMap.h>
+
+#include "gtest/gtest.h"
+#include "sysid/generation/HardwareType.h"
+
+template <size_t S>
+static void TestConversionFunction(
+    std::array<sysid::HardwareType, S> testArray,
+    std::function<sysid::HardwareType(std::string_view)> func) {
+  for (auto& it : testArray) {
+    auto hardwareType = func(it.name);
+    EXPECT_EQ(hardwareType, it);
+  }
+}
```

Comment:
This function could use some documentation to explain what exactly is being tested.